src/libstore/derivations.cc

```diff
@@ -272,7 +272,15 @@ Derivation parseDerivation(const Store & store, std::string && s, std::string_vi
 
 static void printString(string & res, std::string_view s)
 {
-    char buf[s.size() * 2 + 2];
+    char * buf;
+    size_t bufSize = s.size() * 2 + 2;
+    std::unique_ptr<char[]> dynBuf;
+    if (bufSize < 0x10000) {
+        buf = (char *)alloca(bufSize);
+    } else {
+        dynBuf = decltype(dynBuf)(new char[bufSize]);
```

Comment:
```suggestion
        dynBuf = std::make_unique<char[]>(bufSize);
```

Reply:
I try to conform to existing code, such as this pattern taken from `serialise.cc`.
I can change both if @edolstra agrees.